### DISCUSSION TITLE
How to debug a "An error occurred during face recognition, please check the Nextcloud logs."

### DISCUSSION THREAD

**Discussion by beatbesmer** [nextcloud/recognize#684] [Q&A] [answered]
NC25 runs on a shared webhosting with limited access to the shell (chrooted). NC so far is running well, recognize is installed, but i face a "An error occurred during face recognition, please check the Nextcloud logs." and also a photos are not processed ("Face recognition: 5869 Queued files, Last classification: 1.1.1970"), there also are entires in the "jobs" table.

How can i find further informations for a fine grained debugging?

NC config -> loglevel is set to 0
Recognize: 
- node.js -> seems to point correctly to the binary

Or do i need some more privileges (access to system-binaries), which arent available on limited shared webhosting generally?

**Comment by marcelklehr** [comment]
1. If all things are green on the recognize page that's a good sign already.

2. >  "An error occurred during face recognition, please check the Nextcloud logs."

    ...Would indicate there was an error logged somewhere. If you don't see one then that may also not be the issue.

3. Make sure you have cron jobs enabled for the backgroud processing to work.

**Comment by beatbesmer** [comment]
1. From bottom to top:

- `Could not check whether your machine supports native TensorFlow operation`
Is this mandatory?
![image](https://user-images.githubusercontent.com/17173365/221348159-3604bb17-b22f-4bee-8405-51ac8da373a0.png)


- The others have been mentionned already.
![image](https://user-images.githubusercontent.com/17173365/221348215-1db950c5-4002-47da-9618-890d5f17e162.png)

2. Even with activated error_logs, there is nothing recorded in Access / PHP-Logs. Is there any specifig log-file i have to point the hoster towards to look at?

3. CRONs seems triggered. The last_checked of `OCA\Recognize...` is updated regularly

As this runs on a shared-hosting, access to the system is limited. Are there success-stories about working installations on such systems?

**Reply by marcelklehr** [reply]
Nextcloud has its own log, check the docs.

**Reply by marcelklehr** [reply, marked as answer]
> Could not check whether your machine supports native TensorFlow operation

This seems like a likely culprit. If native TensorFlow operation could not be checked,  try enabling WASM mode.

**Comment by beatbesmer** [comment]
Just found the answer
`Privacy
This app does not send any sensitive data to cloud providers or similar services. All image processing is done on your nextcloud machine, using Tensorflow.js running in Node.js, which comes bundled with this app.`
So, node.js is a requirement. No node.js,no recognize :(

**Reply by marcelklehr** [reply]
Node js comes with this app, though.
